app/models/intake.rb

```diff
@@ -439,34 +442,25 @@ def year_before_most_recent_filing_year
   end
 
   def assign_vita_partner!
-    # this should only be called before get_or_create_zendesk_group_id has ever been called
+    # don't re-assign if ZD ticket already exists
+    return if intake_ticket_id.present?
+
     # because we want users to be able to change which group they are routed to up until the ZD ticket has been created,
-    # we don't call get_or_create_zendesk_group_id, we only check whether it has been persisted, and do not save it
-    group_id = zendesk_group_id || determine_zendesk_group_id
-    if group_id
-      partner = VitaPartner.find_by(zendesk_group_id: group_id)
-      raise "unable to determine VITA Partner from zendesk group id: [#{group_id}]" unless partner.present?
-      update(vita_partner_id: partner.id, vita_partner_name: partner.name)
-    end
-  end
+    # we only check whether it has been persisted, and do not save it
+    return nil if zendesk_instance == UwtsaZendeskInstance
 
-  def get_or_create_zendesk_group_id
-    return zendesk_group_id if zendesk_group_id.present?
+    partner, routing_criteria, routing_value = partner_for_source || partner_for_state || partner_for_overflow
```

Comment:
Exciting to see that we are getting all these values back! It might be easier to maintain or work on these in the future with a more explicit output, such as a hash or a minimal "PartnerRoutingResult" class.

Reply:
Non-blocking, but given that we only want two types of results from these methods (`nil` or `Array` of three specific things) think it would be helpful to at least comment if not enforce with a class

Reply:
A light weight Struct can also be a nice in between option until a full blown class with logic is needed.